From 22fae95aff7073da55b2a79055292d320475439b Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 9 May 2017 22:54:16 +0200
Subject: [PATCH] Configure PHP to load the XMLRPC extension if present

This is needed when the system PHP configuration does not already load
the extension.
---
 configure.ac        | 7 ++++++-
 tests/Makefile.am   | 8 +++++++-
 tests/httpd.conf.in | 2 ++
 tests/php.ini.in    | 1 +
 4 files changed, 16 insertions(+), 2 deletions(-)
 create mode 100644 tests/php.ini.in

diff --git a/configure.ac b/configure.ac
index 3df131e67d586522..868b00e7f091a70e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -272,22 +272,26 @@ if test "$have_apache" = 1; then
 
     if test "$have_php" = yes; then
 	AC_MSG_CHECKING([for php-xmlrpc])
-	if $PHP --rf xmlrpc_server_create | grep -q "does not exist"; then
+	if $PHP -d extension=xmlrpc.so --rf xmlrpc_server_create | grep -q "does not exist"; then
 	    have_php_xmlrpc=no
+	    IF_HAVE_PHP_XMLRPC=";"
 	else
 	    have_php_xmlrpc=yes
+	    IF_HAVE_PHP_XMLRPC=""
 	    AC_DEFINE(HAVE_PHP_XMLRPC, 1, [Have php-xmlrpc])
 	fi
 	AC_MSG_RESULT($have_php_xmlrpc)
+	AC_SUBST(IF_HAVE_PHP_XMLRPC)
     fi
 
     if test -f "$APACHE_MODULE_DIR/mod_unixd.so"; then
 	IF_HAVE_MOD_UNIXD=""
     else
 	IF_HAVE_MOD_UNIXD="#"
     fi
     AC_SUBST(IF_HAVE_MOD_UNIXD)
 fi
+AM_CONDITIONAL(HAVE_PHP, test "$have_php" = yes)
 
 AC_PATH_PROG(CURL, curl, no)
 if test "$CURL" != no; then
@@ -415,6 +419,7 @@ AC_CONFIG_FILES([
 	po/Makefile
 	tests/Makefile
 	tests/httpd.conf
+	tests/php.ini
 	examples/Makefile
 	docs/Makefile
 	docs/reference/Makefile
diff --git a/tests/Makefile.am b/tests/Makefile.am
index 81a72cbc5ffd9739..d2986975be11eca2 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -73,24 +73,30 @@ test_data +=			\
 	httpd.conf
 endif
 
+if HAVE_PHP
+test_data +=			\
+	php.ini
+endif
+
 RESOURCES = $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=$(srcdir) --generate-dependencies $(srcdir)/soup-tests.gresource.xml)
 
 soup-tests.gresource: soup-tests.gresource.xml $(RESOURCES)
 	$(AM_V_GEN) $(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=$(srcdir) $<
 
 EXTRA_DIST +=			 \
 	htdigest		 \
 	htpasswd		 \
 	httpd.conf.in		 \
 	index.txt		 \
 	libsoup.supp		 \
+	php.ini.in		 \
 	soup-tests.gresource.xml \
 	test-cert.pem		 \
 	test-key.pem		 \
 	xmlrpc-server.php	 \
 	$(RESOURCES)
 
-DISTCLEANFILES += soup-tests.gresource httpd.conf
+DISTCLEANFILES += soup-tests.gresource httpd.conf php.ini
 
 TESTS_ENVIRONMENT += SOUP_TESTS_IN_MAKE_CHECK=1
 
diff --git a/tests/httpd.conf.in b/tests/httpd.conf.in
index b818c12d198e6618..31c36674a9c5d8c2 100644
--- a/tests/httpd.conf.in
+++ b/tests/httpd.conf.in
@@ -31,6 +31,8 @@ LoadModule proxy_connect_module @APACHE_MODULE_DIR@/mod_proxy_connect.so
 LoadModule ssl_module           @APACHE_SSL_MODULE_DIR@/mod_ssl.so
 @IF_HAVE_MOD_UNIXD@LoadModule unixd_module         @APACHE_SSL_MODULE_DIR@/mod_unixd.so
 
+@IF_HAVE_PHP@PHPIniDir .
+
 DirectoryIndex index.txt
 TypesConfig /dev/null
 AddType application/x-httpd-php .php
diff --git a/tests/php.ini.in b/tests/php.ini.in
new file mode 100644
index 0000000000000000..779c022323b2f55f
--- /dev/null
+++ b/tests/php.ini.in
@@ -0,0 +1 @@
+@IF_HAVE_PHP_XMLRPC@extension=xmlrpc.so
-- 
2.12.2

